### ISSUE TITLE
Which format for DANEBOT_RFC2136_TSIG_SECRET?

### ISSUE THREAD

**Issue by blackbit42** [stuvusIT/danebot#9]
Hi

I am able to generate a TLSA RR with danebot, though the following DDNS update fails.
Output of danebot:
```
Error: DNS request @ <redacted> returned with rcode REFUSED
```

The log of my DNS server (powerdns) has this:
```
Jul 30 13:36:47 ns1 <redacted>[43846]: Packet for '<redacted>' denied: Signature with TSIG key '<redacted>' does not match the expected algorithm (hmac-sha512 / hmac-sha256)
```
or
```
Jul 30 13:30:18 ns1 <redacted>[43846]: Packet for '<redacted>' denied: Signature with TSIG key '<redacted>' failed to validate
```

depending on what I try.

I took a look at the sourcecode of `tsigkeyring.py` but wasn't able to actually figure out in which format to provide DANEBOT_RFC2136_TSIG_SECRET.
My TSIG secret is a `hmac-sha512` key. It's used for other purposes successfully, I know it works.
The secret is 76 characters long. Do I need to base64 encode it? Do I need to supply the algorithm along with it?
Please advise.
Thanks!

**Comment by blackbit42** [comment]
FWIW, this naive patch fixes the issue for me. Danebot is now able to perform the DDNS update and place the TLSA RR.

[danebot.txt](https://github.com/user-attachments/files/16433653/danebot.txt)

github didn't like the file with extension ".patch", hence the ".txt"...
I do not assume this will be a final solution for the issue. If in fact it is, let me know and I'll open a proper PR.

**Comment by haslersn** [comment]
It seems that the TSIG algorithm needs to be passed to the `dns.update.Update(..., keyalgorithm=...)` call. The default is `"HMAC_SHA256"`, so danebot currently only supports that. I created the PR #11 which makes it configurable. Can you check if that works for you?

**Comment by blackbit42** [comment]
Yes, this works. Thank you!

**Comment by haslersn** [comment]
What was the syntax you provided to the `--rfc2136-tsig-algo` argument?

**Comment by blackbit42** [comment]
I ran `danebot` from a shellscript, this is the relevant section:
```
        --rfc2136-tsig-key "$(cat ${KEYFILE} | cut -d ':' -f 1)" \
        --rfc2136-tsig-algo "$(cat ${KEYFILE} | cut -d ':' -f 2)" \
        --rfc2136-tsig-secret "$(cat ${KEYFILE} | cut -d ':' -f 3)" \
```

The field in the file looks like this:
```
$ cat ${KEYFILE} | cut -d ':' -f 2
hmac-sha512
$
```

**Comment by haslersn** [comment]
Fixed by #11
